### DISCUSSION TITLE
Are the metrics for in/out bytes and message-count that are reported by the nats-server (and fed into utilities like nats-top) 100% accurate or are they approximations?

### DISCUSSION THREAD

**Discussion by dsidirop** [nats-io/nats-server#2818] [Q&A] [answered]
Essentially what the subject says. Based on a quick look in the source-code I tend to believe that the data-usage reported by nats-server (and displayed by the 'nats-top' utility) accounts for every single byte and message that goes through NATS.

Is my assumption correct? (just need to be 100% certain)

Thank you for the great product and the immense quality that you guys are putting into it!

**Comment by derekcollison** [comment, marked as answer]
Yes that is correct.

**Comment by ripienaar** [comment]
Be aware it’s per server though - nats top is not cluster aware.

**Reply by dsidirop** [reply]
@ripienaar Thank you for this insight. So if we have a cluster of nats-server (non-ngs) what is the easiest way to get a view of the aggregate traffic that goes through the cluster (as if its a single nats-server)?

**Reply by ripienaar** [reply]
Use something like the prometheus exporter and gather data from all your servers
